src/components/Modal.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import dialogPolyfill from 'dialog-polyfill/dialog-polyfill.js';
+import 'dialog-polyfill/dialog-polyfill.css';
+import './Modal.css';
+
+class Modal extends Component {
+
+  static propTypes = {
+    title: PropTypes.string,
+    reset: PropTypes.func.isRequired,
+    children: PropTypes.object
+  };
+
+  close () {
+    ReactDOM.findDOMNode(this).close();
+  }
+
+  show () {
+    ReactDOM.findDOMNode(this).showModal();
+  }
+
+  componentDidMount () {
+    const dialog = ReactDOM.findDOMNode(this);
+    if (!dialog.showModal) {
+      dialogPolyfill.registerDialog(dialog);
+    }
+
+    if (this.props.title) {
+      setTimeout(this.show.bind(this), 100);
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    if (prevProps.title && !this.props.title) {
```

Comment:
Why not have an explicit `visible` prop instead?

Reply:
Hmm, seems a bit redundant to me. Then we could have the props title & content with a value, but visible to false, and then the modal would be closed but would still have information about title/content, which seems strange to me... what do you think?

Reply:
It doesn't sound intuitive to me to remove a `title` to close the dialog. All other dialog libraries I used for react follow this pattern, have a look here:
https://github.com/reactjs/react-modal

If you want to optimize displaying the dialog instead of setting `isOpen/visible` to `false` you could just remove it from the DOM entirely.

Reply:
OK! Makes sense

I think since we're using HTML5 `<dialog>` it's better to use the native close rather than remove the element (for possibly native animations etc. in the future)

Committed; let me know what you think